oxo.py

```diff
@@ -1,29 +1,159 @@
 class OxoBoard:
+
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.board_rows = 3
+        self.board_columns = 3
+        self.board_array = []
+        # create the board array
+        for x in range(self.board_rows):
+            self.board_array.append([])
+
+            for y in range(self.board_columns):
+                self.board_array[x].append(0)
+
+        self.show()
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+
+        square_content = self.board_array[x][y]
+        return square_content
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+
+        if self.board_array[x][y] == 0:
+            self.board_array[x][y] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+
+        full_columns = 0
+
+        # check if all of the columns are full
+        for x in range(self.board_rows):
+            if 0 in self.board_array[x]:
+                return False
+            else:
+                full_columns += 1
+        if full_columns == self.board_rows:
+            return True
```

Comment:
Can there ever be a situation where the end of the `for` loop is reached but `full_columns` is not equal to `self.board_rows`? (If so then your function would not return a value, which would be a bug)